app/javascript/packs/project_answers.js

```diff
@@ -122,9 +151,34 @@ export async function main() {
           },
          )
         .then((response) => {
+            // User feedback.
+            input.classList.remove('autosave-input-error');
+            inputAlert.innerHTML = "";
+            setTimeout(function(){
+                if (getUnsavedInputs().length === 0) {
+                    changeAutoSaveStatus('All progress saved.', 'autosave-status-success');
+                } else {
+                    changeAutoSaveStatus('Some answers are still unsaved!', 'autosave-status-error');
+                }
+            }, 400); // Wait a split second to change status back so the 'Saving...' message is actually visible
+
+            // Logging.
             honeySpan.addField('response.status', response.status, false);
         })
         .catch((error) => {
+            // User feedback.
+            input.classList.add('autosave-input-error');
+            inputAlert.innerHTML = "Failed to save answer. <a href='#'>Retry?</a>";
+            inputAlert.querySelector('a').onclick = () => { 
+                // clicking the Retry on any of the failed answers retries them all
+                let allUnsavedInputs = getUnsavedInputs();
+                allUnsavedInputs.forEach((input) => {
```

Comment:
Can simplify to
```suggestion
                getUnsavedInputs().forEach((input) => {
```
since we're not re-using this variable anywhere.

Reply:
Actually, can we rename `input` too? It's technically valid, but confusing to override the outer scope with a local scope.